docs/other-topics/hooks.md

```diff
@@ -2,404 +2,253 @@
+|---------------------------------------|-------|------------------------------------------------------------|
+| `beforeDefine`, `afterDefine`         | ❌     | A new `Model` class is being registered                    |
+| `beforeQuery`, `afterQuery`           | ✅     | A SQL query is being run                                   |
+| `beforeBulkSync`, `afterBulkSync`     | ✅     | `sequelize.sync` is called                                 |
+| `beforeConnect`, `afterConnect`       | ✅     | Whenever a new connection to the database is being created |
+| `beforeDisconnect`, `afterDisconnect` | ✅     | Whenever a connection to the database is being closed      |
+
+:::info
+
+All [Model hooks](#model-sequelize-hooks) are also fired on the sequelize instance:
+
+```typescript
+import { Sequelize } from '@sequelize/core';
+
+const sequelize = new Sequelize(/* options */);
+
+// This will be called whenever findAll is called on any model.
+sequelize.hooks.addListener('beforeFindAll', () => {
+  console.log('findAll has been called a model');
+});
+```
+
+:::
+
+### Registering Sequelize hooks
+
+Instance Sequelize hooks can be registered in two ways:
+
+1. Using the `hooks` property of the `Sequelize` instance:
+   ```typescript
+   import { Sequelize } from '@sequelize/core';
+  
+   const sequelize = new Sequelize(/* options */);
+  
+   // highlight-next-line
+   sequelize.hooks.addListener('beforeDefine', () => {
+     console.log('A new Model is being initialized');
+   });
+   ```
+2. Through the `Sequelize` options:
+   ```typescript
+   import { Sequelize } from '@sequelize/core';
+  
+   const sequelize = new Sequelize({
+     // highlight-next-line
+     hooks: {
+       beforeDefine: () => {
+         console.log('A new Model is being initialized');
+       },
+     },
+   });
+   ```
+
+## Model Sequelize Hooks
+
+`Model` classes support the following hooks:
+
+| Hook name                                                                              | Async | Run when                                                                        |
+|----------------------------------------------------------------------------------------|-------|---------------------------------------------------------------------------------|
+| `beforeAssociate`, `afterAssociate`                                                    | ❌     | Whenever an association is declared on the model                                |
+| `beforeSync`, `afterSync`                                                              | ✅     | When `sequelize.sync` or `Model.sync` are called                                |
+| `beforeValidate`, `afterValidate`, `validationFailed`                                  | ✅     | When the model's attributes are being validated (happens in most model methods) |
+| `beforeFind`, `beforeFindAfterExpandIncludeAll`, `beforeFindAfterOptions`, `afterFind` | ✅     | When `Model.findAll` is called                                                  |
```

Comment:
This runs when `Model.findOne` is called as well right?

Reply:
Done